### PR TITLE
Feature/readme index

### PR DESCRIPTION
**Pull request by ajshajib**
I have added permalinks to the tutorial notebooks in the Readme index. I have sorted the categories and the notebooks within individual categories in a roughly logical order. @sibirrer, feel free to suggest any modification in these orders.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,35 +11,35 @@ with interactive Jupyter notebooks in the cloud.
 Index
 -----
 Getting started
- - `Starting guide <s>`_
- - `Units, coordiante system and parameter definitions <s>`_
- - `Handling FITS files <s>`_
+ - `Starting guide <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/GettingStarted/starting_guide.ipynb>`_
+ - `Units, coordiante system and parameter definitions <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/GettingStarted/units_coordinates_parameters_definitions.ipynb>`_
+ - `Handling FITS files <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/GettingStarted/fits_handling_and_extracting_needed_information_from_the_data.ipynb>`_
```
Comment (sibirrer) [inline README.rst]: here perhaps the text appearing in the link can also be improved

File: README.rst [hunk 2]
```diff
@@ -11,35 +11,35 @@ with interactive Jupyter notebooks in the cloud.
...
- - `Modeling a doubly imaged quasar <URL>`_
- - `Modeling a quadruply imaged quasar <URL>`_
- - `Modling multiple bands simultaneously <URL>`_
- - `Source reconstruction with shapelets <URL>`_
- - `Modeling of catalogue data <URL>`_
- - `Cosmic shear with Einstein ring simulations <URL>`_
+ - `Modeling a simple Einstein ring <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/LensModeling/modeling_a_simple_Einstein_ring.ipynb>`_
+ - `Modeling a doubly imaged quasar <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/LensModeling/modeling_a_doubly_imaged_quasar.ipynb>`_
+ - `Modeling a quadruply imaged quasar <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/LensModeling/modeling_a_quadruply_imaged_quasar.ipynb>`_
+ - `Modling multiple bands simultaneously <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/LensModeling/modeling_multiple_bands_simultaneously.ipynb>`_
```
Comment (sibirrer) [inline README.rst]: typo in modeling

File: README.rst [hunk 3]
```diff
@@ -11,35 +11,35 @@ with interactive Jupyter notebooks in the cloud.
...
- - `Simulation API <s>`_
- - `Skypy meets Lenstronomy <s>`_
-Galaxies
- - `Fitting galaxy light profile (like Galfit) <s>`_
- - `Quasar-host decomposition <s>`_
+ - `Simulation API <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/Simulations/simulation_api.ipynb>`_
+ - `Skypy meets Lenstronomy <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/Simulations/skypy_meets_lenstronomy.ipynb>`_
+Galaxy light fitting
+ - `Fitting galaxy light profile (like Galfit) <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/Galaxies/galfitting_with_lenstronomy.ipynb>`_
+ - `Quasar-host decomposition <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/Galaxies/quasar_host_galaxy_decomposition.ipynb>`_
```
Comment (sibirrer) [inline README.rst]: perhaps change text as well to quasar-host galaxy decomposition?

File: README.rst [hunk 4]
```diff
@@ -11,35 +11,35 @@ with interactive Jupyter notebooks in the cloud.
...
 Dark matter substructure
- - `Substructure challenge - a simple example <s>`_
- - `Substructure Challenge - Mock production notebook <s>`_
+ - `Substructure challenge - a simple example <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/DMSubstructure/substructure_challenge_simple_example.ipynb>`_
+ - `Substructure Challenge - Mock production notebook <https://github.com/lenstronomy/lenstronomy-tutorials/blob/main/Notebooks/DMSubstructure/substructure_challenge_mock_production.ipynb>`_
```
Comment (sibirrer) [inline README.rst]: I would use lower case in challenge since it is not a specific challenge

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks great, just had a few minor suggested changes in the text

File: README.rst (hunk 1 above)
Comment (ajshajib) [reply README.rst]: How about now?

File: README.rst (hunk 2 above)
Comment (ajshajib) [reply README.rst]: Fixed.

File: README.rst (hunk 3 above)
Comment (ajshajib) [reply README.rst]: Done.

File: README.rst (hunk 4 above)
Comment (ajshajib) [reply README.rst]: Done.

**Review by sibirrer — APPROVED** [review]

**Pull request merged**